### PR TITLE
Add coherent noise simulation

### PR DESCRIPTION
**Pull request by ascarpel**
Introduces:
1) gen/src/AddCoherentNoise.cxx and gen/inc/WireCellGen/AddCoherentNoise.h main c++ functions for the Coherent noise simulation. Inspired from AddNoise.cxx
2) in wire-cell-toolkit/python/wirecell/sigproc/noise/main.py, persist.py, and icarus.py introduced the loader to convert the noise spectra into an adequate format
 3) configuration file: wire-cell-toolkit/cfg/pgrapher/experiment/icarus/wct-coherent-noise-check.py

### CODE REVIEW

**Comment by wenqiang-gu** [comment]
Hi @ascarpel, I see 26 files changed. Some change is not desired, for example, cfg/pgrapher/experiment/icarus/params.jsonnet . Could you check again?

**Comment by ascarpel** [comment]
@wenqiang-gu, I made some amends to the .jsonnet files following the discussions we had during this week and earlier today. Other files that appear Inopportunely changed have conflicts only because of blank lines or indents. You should instead review more carefully these files here: 
- python/wirecell/sigproc/main.py
- python/wirecell/sigproc/noise/persist.py
- python/wirecell/sigproc/response/persist.py
- python/wirecell/util/main.py
- python/wirecell/util/wires/onesided.py 
I have edited them to include some new functions for the data preparation in ICARUS and also to have them set for Phyton3 sometimes.

File: gen/src/AddNoise.cxx [hunk 1]
```diff
@@ -51,7 +51,7 @@ void Gen::AddNoise::configure(const WireCell::Configuration& cfg)
     m_model = Factory::find_tn<IChannelSpectrum>(m_model_tn);
     m_nsamples = get<int>(cfg,"nsamples",m_nsamples);
     m_rep_percent = get<double>(cfg,"replacement_percentage",m_rep_percent);
-
```
[the remark this answers is not in the document]
Comment (ascarpel) [reply gen/src/AddNoise.cxx:54]: Ok, thank you! this is very useful. I can check all the main differences and restore the previous format.

File: gen/src/AddCoherentNoise.cxx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      }
+
+      if( random_amplitudes.size() != spec.size() ){
+        random_amplitudes.resize(spec.size());
+      }
+
+      // Create the ampls vector and multiply for the phase
+      WireCell::Waveform::compseq_t noise_freq(spec.size(),0);
+
+      for (int i=0;i<int(spec.size());i++)
```
[the remark this answers is not in the document]
Comment (ascarpel) [reply gen/src/AddCoherentNoise.cxx]: Thanks, I have updated following this style

**Comment by HaiwangYu** [comment]
I tested this branch util bee1aab with a `nf-sp-dnnroi` chain. The results are identical with before this PR.

**Comment by wenqiang-gu** [comment]
I can successfully run the coherent simulation with the configuration `wct-coherent-noise.jsonnet`. I suggest we merge this pull request although we certainly need a better channel grouping strategy for the use of other experiments. [An issue](https://github.com/WireCell/wire-cell-toolkit/issues/77#issue-602198711) has been created for this purpose. @brettviren What you think?

**Comment by brettviren** [comment]
It's a good plan, @wenqiang-gu.  Thank you and @HaiwangYu for testing and of course @ascarpel for adding this!

**Pull request merged**